### PR TITLE
Add handling for prompts returned from zypper.

### PR DESCRIPTION
**Pull request by skazi0**
Sometimes zypper has some pending prompt which will be shown on every
invocation (e.g. repo key confirmation). In such case, repocheck fails
silently with 500 and log does not contain much useful information.

This change adds proper handling for such cases with the prompt text
returned as a response from the call.

### CODE REVIEW

**Comment by jsuchome** [comment]
Hm, can't we just pass some argument to zypper to always confirm any questions?

**Comment by skazi0** [comment]
Hmm, `--gpg-auto-import-keys` would (as the name implies) import the keys without asking which is not what every user would want. Same for `--auto-agree-with-licenses`.

The only one which makes sense is `--non-interactive` but it will silently reject the signing key and silenty skip the repo. If this is what we want then it will be an easier fix.

File: crowbar_framework/app/models/api/upgrade.rb [hunk 1]
```diff
@@ -90,12 +90,21 @@ def adminrepocheck
           if zypper_stream["message"] =~ /^System management is locked/
             return {
               status: :service_unavailable,
-              message: I18n.t(
+              error: I18n.t(
```
Comment (jsuchome) [inline crowbar_framework/app/models/api/upgrade.rb]: I know it's not part of your change, we have another problem, that even the normal repo check performed bellow does not return "error"...

Comment (jsuchome) [inline crowbar_framework/app/models/api/upgrade.rb]: OK, maybe not. I meant the result of admin repo check when the repos are actually not there, but we do not have to treat it like an error, just a special kind of result.

Comment (skazi0) [inline crowbar_framework/app/models/api/upgrade.rb]: Yes, these are not errors. They are just 'negative' result of checks.

**Pull request merged**